docs/articles/documentation/using-testcafe/programming-interface/Runner.md

```diff
@@ -0,0 +1,223 @@
+
+#### Example
+
+```js
+runner.screenshots('/home/user/screenshots/', true);
+```
+
+### <a class="anchor" name="reporter"></a>reporter(name, outStream]) → this
+
+Configures the test runner's reporting feature.
+
+Parameter                | Type                        | Description                                     | Default
+------------------------ | --------------------------- | ----------------------------------------------- | --------
+`name`                   | String                      | The name of the reporter to use.
+`outStream` *(optional)* | Writable Stream implementer | The stream to which the report will be written. | `stdout`
+
+Reporters are plugins used to output test run reports in a certain format.
+
+TestCafe ships with the following reporters:
+
+* [spec](https://github.com/DevExpress/testcafe-reporter-spec) - used by default,
+* [list](https://github.com/DevExpress/testcafe-reporter-list),
+* [minimal](https://github.com/DevExpress/testcafe-reporter-minimal),
+* [xUnit](https://github.com/DevExpress/testcafe-reporter-xunit),
+* [JSON](https://github.com/DevExpress/testcafe-reporter-json).
+
+#### Examples
+
+##### Specifying the reporter
+
+```js
+runner.reporter("minimal");
+```
+
+##### Saving the report to a file
+
+```js
+var stream = fs.createWriteStream('./report.xml');
+
+runner
+    .reporter("xunit", stream)
+    .run()
+    .then(() => stream.end());
+```
+
+##### Handle the report programmatically
+
+```js
+runner.reporter("json", {
+    write: (chunk, encoding, next) => {
+        doSomething(chunk);
+        next();
+    }
+});
+```
+
+You can also build your own reporter. Use a [dedicated Yeoman generator](https://github.com/DevExpress/generator-testcafe-reporter) to scaffold out a custom reporter plugin.
+
+### <a class="anchor" name="run"></a>run(options) → Promise\<Number\>
+
+Runs tests according to the current configuration. Returns the number of failed tests.
+
+You can pass the following options to this function.
+
+Parameter                | Type    | Description                                                                                                                                                                                            | Default
+------------------------ | ------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | -------
+`options.skipJsErrors`   | Boolean | Defines whether to continue running a test after a JavaScript error occurs on a page (`true`), or consider such a test failed (`false`).                                                               | `false`
+`options.quarantineMode` | Boolean | Defines whether to mark a test failed after the first unsuccessful run (`false`), or send it to the quarantine where it will be run again several times and classified as failed or unstable (`true`). | `false`
```

Comment:
Worth mention that any test that has different results in quarantine test run despite it's final results will be marked as `unstable` in report. So it can be failed and unstable or passed and unstable. Final result calculated from the majority of the results. And I think we can mention this article here: http://martinfowler.com/articles/nonDeterminism.html